modin/engines/dask/pandas_on_dask_delayed/block_partitions.py

```diff
@@ -13,3 +17,57 @@ class DaskBlockPartitions(BaseBlockPartitions):
 
     def __init__(self, partitions):
         self.partitions = partitions
+
+    def _get_partitions(self):
+        # We don't filter fot the laziness of Dask Delayed
+        return self._partitions_cache
+
+    def _set_partitions(self, new_partitions):
+        self._partitions_cache = new_partitions
+
+    partitions = property(_get_partitions, _set_partitions)
+
+    def to_pandas(self, is_transposed=False):
+        """Convert this object into a Pandas DataFrame from the partitions.
+
+        Args:
+            is_transposed: A flag for telling this object that the external
+                representation is transposed, but not the internal.
+
+        Returns:
+            A Pandas DataFrame
+        """
+        # In the case this is transposed, it is easier to just temporarily
+        # transpose back then transpose after the conversion. The performance
+        # is the same as if we individually transposed the blocks and
+        # concatenated them, but the code is much smaller.
+        if is_transposed:
+            return self.transpose().to_pandas(False).T
+        else:
+            retrieved_objects = [
+                list(dask.compute(*[obj.dask_obj for obj in part]))
+                for part in self.partitions
+            ]
```

Comment:
In general it's nicer to call dask.compute on as much as you can at once.  You might want to pull the dask.compute call outside of the list comprehension here.

Reply:
I will do that, thanks! This is a 2D array, so I will just flatten the entire thing for the `dask.compute`.